### ISSUE TITLE
CTests show PASS when segfaults occur

### ISSUE THREAD

**Issue by sbrus89** [E3SM-Project/Omega#87]
Not sure if anyone else has noticed this, but I often see that a test shows as passed even when seg faults occur for that test in the log file. I'm assuming this is because there is no explicit `FAIL` expression found. For example:
```
      Start  8: HORZOPERATORS_PLANE_TEST
 8/15 Test  #8: HORZOPERATORS_PLANE_TEST .........   Passed    3.61 sec
```
when in `Testing/Temporary/LastTest.log ` there are seg fault errors:
```
8/15 Testing: HORZOPERATORS_PLANE_TEST
8/15 Test: HORZOPERATORS_PLANE_TEST
Command: "/usr/bin/srun" "-n 8" "--cpu-bind=cores" "--" "./testHorzOperatorsPlane.exe"
Directory: /ccs/home/brus/project/Omega/components/omega/build_cpu/test
"HORZOPERATORS_PLANE_TEST" start time: Jun 03 17:36 EDT
Output:
----------------------------------------------------------
[2024-06-03 17:36:58.205] [info] [HorzOperatorsTest.cpp:217] OperatorsTest: Divergence PASS 
[2024-06-03 17:36:58.205] [info] [HorzOperatorsTest.cpp:217] OperatorsTest: Divergence PASS 
[2024-06-03 17:36:58.205] [info] [HorzOperatorsTest.cpp:217] OperatorsTest: Divergence PASS 
[2024-06-03 17:36:58.205] [info] [HorzOperatorsTest.cpp:217] OperatorsTest: Divergence PASS 
[2024-06-03 17:36:58.205] [info] [HorzOperatorsTest.cpp:217] OperatorsTest: Divergence PASS 
[2024-06-03 17:36:58.205] [info] [HorzOperatorsTest.cpp:217] OperatorsTest: Divergence PASS 
[2024-06-03 17:36:58.205] [info] [HorzOperatorsTest.cpp:217] OperatorsTest: Divergence PASS 
[2024-06-03 17:36:58.205] [info] [HorzOperatorsTest.cpp:217] OperatorsTest: Divergence PASS 
srun: error: frontier08975: tasks 0-4,6-7: Segmentation fault
srun: Terminating StepId=1998017.46
slurmstepd: error: *** STEP 1998017.46 ON frontier08975 CANCELLED AT 2024-06-03T17:36:58 ***
srun: error: frontier08975: task 5: Terminated
srun: Force Terminated StepId=1998017.46
<end of output>
Test time =   0.92 sec
----------------------------------------------------------
Test Pass Reason:
Required regular expression found. Regex=[PASS]
"HORZOPERATORS_PLANE_TEST" end time: Jun 03 17:36 EDT
"HORZOPERATORS_PLANE_TEST" time elapsed: 00:00:00
----------------------------------------------------------
```
Is there anything we can do to catch this behavior and flag the test as `FAIL`?

**Comment by xylar** [comment]
That's quite concerning!  Shouldn't we FAIL in any situation where PASS is not detected?

**Comment by sbrus89** [comment]
I don't think that would quite prevent the situation above, since as long as at least one test passes it would still result in an overall PASS (even though a segfault occurs before all tests complete).

**Comment by xylar** [comment]
I was referring to individual tests, rather than the test suite.  Is there not a mechanism to print `FAIL` when a segfault occurs on an individual test?

**Comment by mwarusz** [comment]
Looking at the docs for [`PASS_REGULAR_EXPRESSION`](https://cmake.org/cmake/help/latest/prop_test/PASS_REGULAR_EXPRESSION.html), I think it might be because we set this property. The docs say that when it is set exit codes are ignored.

Now, these docs also say that segfaults should still fail the tests. However, it is likely that `CMake` doesn't catch that if `srun`/`mpirun` is used, since if a segfault happens it happens in a spawned process.

One thing that might work is to not use `PASS_REGULAR_EXPRESSION` and simple treat the tests that don't fail as passed. Then both `FAIL` and non-zero exit codes would cause test failures. Segfaults might get handled then, since `srun`/`mpirun` errors when a segfault happens.

**Comment by sbrus89** [comment]
I removed `PASS_REGULAR_EXPRESSION`  and tried it out on [`010ef4dec`](https://github.com/E3SM-Project/Omega/commit/010ef4dece27cc8339888020f07dc214df84aef1) and it seems to work as @mwarusz was hoping it would with regard to catching `srun` errors from segfaults. I'll open a PR and others can chime in on whether there are any potential downsides to this approach.

**Comment by xylar** [comment]
I believe this was fixed in #93.  Please reopen if I'm mistaken.
